src/util.ts

```diff
@@ -844,3 +844,45 @@ export function makeMemoizedByteOffsetConverter(buffer: Buffer): (byteOffset: nu
 	};
 }
 
+export class TempFileProvider {
+	private constructor() {}
+	private static _instance: TempFileProvider;
+
+	static get instance() {
+		if (!TempFileProvider._instance) {
+			TempFileProvider._instance = new TempFileProvider();
+		}
+		return TempFileProvider._instance;
+	}
+
+	private store: vscode.Memento;
```

Comment:
let's call this the `globalState` so that when code reading we know exactly where the data is stored